web/client/components/shapefile/ShapefileUploadAndStyle.jsx

```diff
@@ -157,9 +157,9 @@ class ShapeFileUploadAndStyle extends React.Component {
             }, []);
             this.props.onShapeChoosen(ls);
             this.props.shapeLoading(false);
-        }).catch((e) => {
+        }).catch(() => {
             this.props.shapeLoading(false);
-            this.props.onShapeError(e.message || e);
+            this.props.onShapeError(<Message msgId="shapefile.error.addShape"/>);
```

Comment:
this sends a react component in the state. You should set the message id in the state and use the `<Message` in the error div component.